### PR TITLE
Enable s3up inside docker container

### PR DESCRIPTION
**Pull request by abal5**
Use docker with internal s3cmd to upload artifacts, replacing the reliance on physical slaves which sometimes have misconfigured s3cmd.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -66,7 +66,11 @@ pipeline {
                                     nvidia-docker exec ${CONTAINER_NAME} ln -s /data ./data
                                     nvidia-docker exec ${CONTAINER_NAME} rm -rf open_data
                                     nvidia-docker exec ${CONTAINER_NAME} ln -s /open_data ./open_data
-                                    nvidia-docker exec ${CONTAINER_NAME} bash -c '. /h2oai_env/bin/activate; ./scripts/gitshallow_submodules.sh; make ${env.MAKE_OPTS} AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID} AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY} fullinstalljenkins ; rm -rf build/VERSION.txt ; make build/VERSION.txt'
+                                    nvidia-docker exec ${
+                                CONTAINER_NAME
+                            } bash -c '. /h2oai_env/bin/activate; ./scripts/gitshallow_submodules.sh; make ${
```
Comment (navdeep-G) [inline Jenkinsfile]: Just cosmetics, but the virtualenv should probably be `h2o4gpu_env`

File: Jenkinsfile [hunk 2]
```diff
@@ -142,43 +146,45 @@ pipeline {
 
         stage('Publish to S3') {
             agent {
-                label "linux && !mr-dl16"
+                label "docker"
```
Comment (navdeep-G) [inline Jenkinsfile]: What is this doing now?

File: Jenkinsfile [hunk 3]
```diff
@@ -142,43 +146,45 @@ pipeline {
...
+                        if (isRelease()) {
+                            s3up {
+                                localArtifact = 'src/interface_py/dist/h2o4gpu-*-py36-none-any.whl'
+                                artifactId = "h2o4gpu"
+                                majorVersion = _majorVersion
+                                buildVersion = _buildVersion
+                                keepPrivate = false
+                                remoteArtifactBucket = "s3://artifacts.h2o.ai/releases/stable"
+                            }
+                            sh "s3cmd setacl --acl-public s3://artifacts.h2o.ai/releases/stable/ai/h2o/h2o4gpu/${versionTag}/h2o4gpu-${versionTag}-py36-none-any.whl"
```
Comment (navdeep-G) [inline Jenkinsfile]: Would we need another for py35?

File: Jenkinsfile (hunk 2 above)
Comment (abal5) [reply Jenkinsfile]: it's using the dockerfile from the repo, which has s3cmd configured inside it already 

the archive and upload step should now run inside the container and upload to s3; this is the main purpose of this PR

Comment (navdeep-G) [reply Jenkinsfile]: Makes sense and a good way to go.

**Comment by navdeep-G** [comment]
LGTM

**Comment by abal5** [comment]
will merge once it passes with all changes from master, should fix up the s3up issue

**Comment by pseudotensor** [comment]
Getting:


CONSOLE OUTPUT
[...truncated 5.46 MB...]
[Pipeline] {
[Pipeline] timeout
Timeout set to expire in 1 min 40 sec
[Pipeline] {
[Pipeline] checkout
> git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
> git config remote.origin.url https://github.com/h2oai/h2o4gpu.git # timeout=10
Fetching upstream changes from https://github.com/h2oai/h2o4gpu.git
> git --version # timeout=10
using GIT_ASKPASS to set credentials
> git fetch --tags --progress https://github.com/h2oai/h2o4gpu.git +refs/heads/s3docker:refs/remotes/origin/s3docker
ERROR: Error fetching remote repo 'origin'
hudson.plugins.git.GitException: Failed to fetch from https://github.com/h2oai/h2o4gpu.git
at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:817)
at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1084)
at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1115)
at org.jenkinsci.plugins.workflow.steps.scm.SCMStep.checkout(SCMStep.java:113)
at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:85)
at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:75)
at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1$1.call(AbstractSynchronousNonBlockingStepExecution.java:47)
at hudson.security.ACL.impersonate(ACL.java:260)
at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1.run(AbstractSynchronousNonBlockingStepExecution.java:44)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
at java.lang.Thread.run(Thread.java:745)
Caused by: hudson.plugins.git.GitException: Command "git fetch --tags --progress https://github.com/h2oai/h2o4gpu.git +refs/heads/s3docker:refs/remotes/origin/s3docker" returned status code 128:
stdout:
stderr: fatal: Couldn't find remote ref refs/heads/s3docker
at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1924)
at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandWithCredentials(CliGitAPIImpl.java:1643)
at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.access$300(CliGitAPIImpl.java:71)
at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$1.execute(CliGitAPIImpl.java:352)
at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler$1.call(RemoteGitImpl.java:153)
at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler$1.call(RemoteGitImpl.java:146)
at hudson.remoting.UserRequest.perform(UserRequest.java:181)
at hudson.remoting.UserRequest.perform(UserRequest.java:52)
at hudson.remoting.Request$2.run(Request.java:336)
at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
at java.lang.Thread.run(Thread.java:748)
at ......remote call to mr-dl11(Native Method)
at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1554)
at hudson.remoting.UserResponse.retrieve(UserRequest.java:281)
at hudson.remoting.Channel.call(Channel.java:839)
at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.execute(RemoteGitImpl.java:146)
at sun.reflect.GeneratedMethodAccessor612.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.invoke(RemoteGitImpl.java:132)
at com.sun.proxy.$Proxy104.execute(Unknown Source)
at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:815)

**Comment by pseudotensor** [comment]
http://mr-0xc1:8080/job/h2o4gpu/job/s3docker/5/

**Pull request merged**